auth/oras-go/client.go

```diff
@@ -0,0 +1,42 @@
+package orasgo
+
+import (
+	"context"
+	"crypto/tls"
+	"net/http"
+
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+// client option struct
+type ClientOptions struct {
+	Credential      auth.Credential
+	CredentialStore *Store
+	Debug           bool
+}
+
+func NewClient(opts ClientOptions) remote.Client {
+	client := &auth.Client{
+		Client: &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					InsecureSkipVerify: false,
+				},
+			},
+		},
+	}
```

Comment:
We need to construct the `auth.Client` carefully with the context of ACR.

Reply:
`ForceAttemptOAuth2: true` causes 400 bad request when logging in.

Reply:
`ForceAttemptOAuth2: true` is a known issue, to be addressed later.